Library/Homebrew/download_strategy.rb

```diff
@@ -119,6 +120,10 @@ def fetch
     else
       clone_repo
     end
+
+    if @ref_type == :tag && @revision && !current_revision?
```

Comment:
I think it reads funny to check that `@revision` is not nil but then hide the comparison behind another method. Is it logical to ask the download strategy if itself is the `current_revision?`, or would it make more sense to ask the download strategy _for_ the current revision and then compare it with given revision?